rfcs/react-components/non-core-components.md

```diff
@@ -0,0 +1,388 @@
+
+#### Faster than core components
+
+Breaking changes for non-core components can occur more frequently than for core components. The proposal does not
+mean that breaking changes will be taken lightly, or that there will be a requirement to make breaking changes often.
+
+##### Pros
+
+- 👍 Bleeding edge improvements for partners
+- 👍 Specialized components might not have widespread use and can be upgraded in a few instances
+
+##### Cons
+
+- 👎 Cost of breaking changes has not changed
+- 👎 Just because we make breaking changes does not mean users will upgrade earlier
+- 👎 Need to handle urgent bugfixes for most past versions - need a clear support SLA
+
+### Choosing a repo for
+
+#### Current monorepo
+
+Non-core packages should still live in the monorepo.
+
+##### Pros
+
+- 👍 Easier setup
+- 👍 Nothing new to learn
+- 👍 Issue reporting path does not change
+
+##### Cons
+
+- 👎 We might not want the same release cadence - releasing a fix for a non-core component would release all of core
+- 👎 Extra load on the CI as more non-core components are needed
+- 👎 Not possible to have more separation from core components without lots of tooling hacks
+
+#### Separate monorepo
+
+We can create a new monorepo for our non-core packages and depend on core packages from NPM.
+
+##### Pros
+
+- 👍 We can have our ideal DX
+- 👍 Release cadence does not need to depend on core
+- 👍 Developing our with our public API - we eat our own dogfood
+
+##### Cons
+
+- 👎 Needs effort to setup
+- 👎 Issue reporting becomes more unclear
+- 👎 Requires linking to use latest core code
+- 👎 Manual/Automated bumps required to keep packages up to date
+
+### Package scope
+
+#### @fluentui scope
+
+Non-core packages would be published under the same `@fluentui/` NPM scope.
+
+##### Pros
+
+- 👍 No need to create and manage another organization
+- 👍 Easier discoverability for OSS users and internal partners
+
+##### Cons
+
+- 👎 Not clear that there is a difference between non-core components and core components/utilities
+- 👎 Harder for us to tell the difference between non-core and core
+
+#### New package scope
+
+Creates a new package scope, `@fluentui-banana/` (name pending) for all components/packages that are non-core.
+
+##### Pros
+
+- 👍 Very clear differentiation between core and non-core
+- 👍 Creates a clear boundary for FUI team and partners to collaborate
+- 👍 Clear path to upgrade/downgrade packages between core and non-core (i.e. experiments/compat)
+
+##### Cons
+
+- 👎 Need to maintain a new NPM organization and credentials
+- 👎 Initially could cause some confusion if we don't communicate this well
+
+### How users should consume components
+
+We should keep the same architecture we currently have for our core components: one package per component. What should
+be decided here is how users consume these components.
+
+#### Suite packages
+
+Structure non-core components into a new suite package similar to `@fluentui/react-components`.
```

Comment:
in FluentAI we plan to create a suite package for sure. For a group of related components, this certainly makes the most sense.

Reply:
this tends to be tricky. maybe similar group of extension that covers particular area might be under common suite but as we will have contributions from various areas/partners this might never happen. 

maybe we can follow a rule like:

1. start as package
2. if there are packages that share same/similar domain we can spin a new suite package that re-exports those
  - ideally this suite should be maintained by the same team. by that it doesn't mean that it can contain only sub-packages from the same team, basically owner of majority of that domain should be the main owner/maintainer of the X suite

Reply:
I'll remove this section from the RFC since it seems like there's no easy 'catch-all' rule and we'll probably be looking at all cases individually anyway